examples/uart-dma.rs

```diff
@@ -0,0 +1,245 @@
+
+        let (tx, mut rx) = uart3.split();
+
+        rx.listen_idle();
+
+        cortex_m::interrupt::free(|cs| *G_UART3_TX.borrow(cs).borrow_mut() = Some(tx));
+
+        cortex_m::interrupt::free(|cs| {
+            *G_UART3_BUFFER.borrow(cs).borrow_mut() = Some(Buffer::new());
+        });
+        // Initialize and start DMA stream
+        let mut rx_transfer = Transfer::init_peripheral_to_memory(
+            dma1.1,
+            rx,
+            rx_buffer1,
+            None,
+            DmaConfig::default()
+                .memory_increment(true)
+                .fifo_enable(true)
+                .fifo_error_interrupt(true)
+                .transfer_complete_interrupt(true),
+        );
+
+        rx_transfer.start(|_rx| {});
+
+        cortex_m::interrupt::free(|cs| *G_TRANSFER.borrow(cs).borrow_mut() = Some(rx_transfer));
+
+        // Enable interrupt
+        unsafe {
+            cortex_m::peripheral::NVIC::unmask(Interrupt::USART3);
+            cortex_m::peripheral::NVIC::unmask(pac::Interrupt::DMA1_STREAM1);
+        }
+    }
+
+    // to test this, connect the RX and TX pins together for a loopback
+    // - so you can listen to what you just sent.
+
+    uart3_write(b"hello world\r").expect("Failed to write to UART3");
+    let _response = uart3_read_until(b'\r').expect("Failed to read from UART3");
+    // ... do something with response
+
+    loop {
+        cortex_m::asm::nop();
+    }
+}
+
+#[interrupt]
+#[allow(non_snake_case)]
+fn USART3() {
+    cortex_m::interrupt::free(|cs| {
+        if let Some(transfer) = G_TRANSFER.borrow(cs).borrow_mut().as_mut() {
+            if transfer.is_idle() {
+                // Calc received bytes count
+                let bytes_count = UART_BUFFER_SIZE - transfer.number_of_transfers() as usize;
+                unsafe {
+                    let mut buffer = [0; UART_BUFFER_SIZE];
+                    match transfer.next_transfer(&mut RX_UART3_BUFFER) {
+                        Ok((b, _)) => buffer = *b,
+                        Err(_err) => {}
+                    }
+                    if let Some(ring_buffer) = G_UART3_BUFFER.borrow(cs).borrow_mut().as_mut() {
+                        for i in 0..bytes_count {
+                            ring_buffer.push(buffer[i]);
+                        }
+                    }
+                }
+            }
+            transfer.clear_idle_interrupt();
+        }
+    });
+}
+
+#[interrupt]
+#[allow(non_snake_case)]
+fn DMA1_STREAM1() {
+    cortex_m::interrupt::free(|cs| {
+        if let Some(transfer) = G_TRANSFER.borrow(cs).borrow_mut().as_mut() {
+            // Its important to clear fifo errors as the transfer is paused until it is cleared
+            if transfer.flags() == DmaFlag::FifoError {
+                transfer.clear_fifo_error();
+            }
+            if transfer.flags() == DmaFlag::TransferComplete {
+                transfer.clear_transfer_complete();
+            }
```

Comment:
It is faster to clear without check, is not it?
```suggestion
            transfer.clear_flags(DmaFlag::FifoError | DmaFlag::TransferComplete);
```

Reply:
Agreed, so I guess it does not matter if we clear it even though the transfer is not yet fully complete (`HalfTransfer`) ?